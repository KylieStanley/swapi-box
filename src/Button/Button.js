import React from 'react'
import { NavLink } from 'react-router-dom'
import './Button.scss'
import '../Menu/Menu.scss'
import PropTypes from 'prop-types'


const Button = (props) => {
  return (
    <li className="menu-item">
      <NavLink to={`/main/${props.name}`} className="nav-btn">
        {props.name}
      </NavLink>
    </li>
  )
}

export default Button

Button.propTypes = {
  name: PropTypes.string.isRequired
}
