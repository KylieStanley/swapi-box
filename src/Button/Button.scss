.active {
  box-shadow: 0 0 8px #03b9ffb0, 0 0 16px #03b9ffb0, 0 0 24px #03b9ffb0, 0 0 32px #03b9ffb0, 0 0 40px #03b9ffb0, 0 0 48px #03b9ffb0;
  }
