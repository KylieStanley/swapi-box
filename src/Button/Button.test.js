import React from 'react'
import { shallow, mount } from 'enzyme'
import Button from './Button.js'

describe('Button', () => {
  it('should match the snapshot', () => {
    const mockSelection = 'People'
    const mockName = 'Vehicles'
    const wrapper = shallow(<Button currentSelection={ mockSelection } name={ mockName }/>)
    
    expect(wrapper).toMatchSnapshot()
  })
})