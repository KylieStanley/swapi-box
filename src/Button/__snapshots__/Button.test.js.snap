// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    currentSelection="People"
    name="Vehicles"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        href="#menu"
      >
        <span>
          Vehicles
        </span>
      </a>,
      "class": " menu-item",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          Vehicles
        </span>,
        "href": "#menu",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "span",
      },
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          href="#menu"
        >
          <span>
            Vehicles
          </span>
        </a>,
        "class": " menu-item",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            Vehicles
          </span>,
          "href": "#menu",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "span",
        },
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
